distributed/process.py

```diff
@@ -126,9 +132,9 @@ def stop_thread(q):
         self._thread_finalizer = weakref.finalize(self, stop_thread, q=self._watch_q)
         self._thread_finalizer.atexit = False
 
-    def _on_exit(self, exitcode):
+    def _on_exit(self, exitcode: int) -> None:
         # Called from the event loop when the child process exited
-        self._process = None
+        self._process = None  # type: ignore[assignment]
         if self._exit_callback is not None:
             self._exit_callback(self)
```

Comment:
the exit_callback is called here - and if `set_exit_callback` was a coroutine function results in an unawaited coroutine warning